### PR TITLE
Update BTCPayService.php

### PR DESCRIPTION
**Pull request by dni**
as mentioned in issue #5

### CODE REVIEW

File: Model/BTCPay/BTCPayService.php [hunk 1]
```diff
@@ -205,6 +205,13 @@ public function createInvoice(Order $order): array
 
         $ba = $order->getBillingAddress();
         $sa = $order->getShippingAddress();
+      
+        // if there is no shipping address, use the billing address
+        // this happens if all products in an order are virtual or downloadable
+        // and no shipping is required
+        if (!$sa) {
```
Comment (woutersamaey) [inline Model/BTCPay/BTCPayService.php]: Please use `$order->isVirtual()`

**Review by woutersamaey — CHANGES_REQUESTED** [review]

File: Model/BTCPay/BTCPayService.php (hunk 1 above)
Comment (dni) [reply Model/BTCPay/BTCPayService.php]: i dont know why, but this doesnt work for me, it doesnt trigger, $sa is still null in my case and throws.

Comment (woutersamaey) [reply Model/BTCPay/BTCPayService.php]: That does not sound healthy. Can you debug? Might be something with your store then...
if all items in the order are virtual (i.e. not delivery needed), it should say the order itself is virtual and there won't be a shipping address presented at checkout.

My solution is Magento best practise and we should aim to do it this way.

Comment (dni) [reply Model/BTCPay/BTCPayService.php]: `Invalid method Magento\Sales\Model\Order\Interceptor
▶ 268                 \Magento\Framework\Profiler::stop('magento');                   |::isVirtual

Comment (dni) [reply Model/BTCPay/BTCPayService.php]: screenshot-1639390127

Comment (dni) [reply Model/BTCPay/BTCPayService.php]: somehow data in the order is_virtual is 1 but the function retursn false... im very confused, maybe the isVirtual() doesnt exist anymore?

**Comment by dni** [comment]
so it works for me now with `if ($order->getIsVirtual() || $order->getIsDownloadable()) {`

im using magento 2.4.3

**Comment by woutersamaey** [comment]
Superb, tnx!

**Comment by dni** [comment]
np, i'm very happy, that actually some1 is working on that plugin, i'm going to use it more in the future.

**Pull request merged**